### ISSUE TITLE
`unneeded_escaping` false positive when variable later used in a tuple

### ISSUE THREAD

**Issue by rgoldberg** [realm/SwiftLint#6621]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

`unneeded_escaping` false positive when variable later used in a tuple.

```swift
private import os

final class Lazy<Value: Sendable>: Sendable {
	private let state: OSAllocatedUnfairLock<(value: Value?, initializer: (@Sendable () -> Value)?)>

	var value: Value {
		state.withLock { state in
			state.value ?? {
				let value = state.initializer!() // swiftlint:disable:this force_unwrapping
				state.value = value
				state.initializer = nil
				return value
			}()
		}
	}

	init(_ initializer: @escaping @Sendable () -> Value) { // swiftlint:disable:this unneeded_escaping
		state = .init(initialState: (value: nil, initializer: initializer))
	}
}
```

```shell
swiftlint
```

### Environment

* SwiftLint version: 0.63.2
* Xcode version: Xcode 26.4.1 / Build version 17E202
* Installation method used: Homebrew
* Configuration file:

```yml
only_rules:
  - unneeded_escaping
```

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
